### ISSUE TITLE
LabelSelector is not being used to schedule finetuning job/pods on selected nodes

### ISSUE THREAD

**Issue by varkat77** [kaito-project/kaito#1428]
**Describe the bug**
I am trying to schedule fine tuning on BYO nodes and observed LabelSelector is not used to schedule finetuning job/pods on selected nodes. When fine tuning workspace is created, it creates a new node pool with selected SKU and pod is scheduled on other node pools that has matching SKU machines available with different labels.

On AKS, FineTuning workspace has label selectors but I don’t see any nodeaffinity or node selectors settings in the fine-tuning job or pod. Seems tuning preset doesn’t take into account label selectors [https://github.com/kaito-project/kaito/blob/9b79ccc4334f74ef5c312b646703b423ec802f6e/pkg/workspace/tuning/preset_tuning.go](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fkaito-project%2Fkaito%2Fblob%2F9b79ccc4334f74ef5c312b646703b423ec802f6e%2Fpkg%2Fworkspace%2Ftuning%2Fpreset_tuning.go&data=05%7C02%7Cv-vakataria%40microsoft.com%7C29acbe9f4d7b460fdcf608dde1c17b8a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638914943300039017%7CUnknown%7CTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7C0%7C%7C%7C&sdata=x%2FghMS63ViABQAczirfB3BzNqFKmBCMzmBBM6ScdDUM%3D&reserved=0)

Inference workspaces are hosted correctly as per label selectors, only finetuning workspace has this issue.

**Steps To Reproduce**
Pre-provision node pool for fine tuning, Use LabelSelectors to schedule on pre-provisioned node pool. If there are other node pools with requested SKU then pods will be scheduled randomly on any of these pools ignoring label selectors.

**Expected behavior**
Fine tuning should be scheduled on matching label selectors node pool only
